src/client.rs

```diff
@@ -3,9 +3,14 @@ use crate::{
     flight::SqlFlightClient,
     tls::new_tls_flight_channel,
 };
+use arrow::error::ArrowError;
+use arrow::record_batch::RecordBatch;
 use arrow_flight::decode::FlightRecordBatchStream;
+use spice_util::{fibonacci_backoff::FibonacciBackoffBuilder, retry, RetryError};
 use tonic::transport::Channel;
 
+const MAX_RETRIES: usize = 5;
```

Comment:
Should add method to control retries and be consistent with default retries num (3), similar to go client
https://github.com/spiceai/gospice?tab=readme-ov-file#connection-retry

```
Connection retry
The SpiceClient implements connection retry mechanism (3 attempts by default). The number of attempts can be configured via SetMaxRetries:

spice := NewSpiceClient()
spice.SetMaxRetries(5) // Setting to 0 will disable retries
Retries are performed for connection and system internal errors. It is the SDK user's responsibility to properly handle other errors, for example RESOURCE_EXHAUSTED (HTTP 429).
```